service/discovery/discovery.go

```diff
@@ -186,6 +191,14 @@ func NewService(opts ...ServiceOption) *Service {
 		o(s)
 	}
 
+	// Default to an in-memory storage, if nothing was explicitly set
+	if s.storage == nil {
+		s.storage, err = inmemory.NewStorage()
+		if err != nil {
+			log.Errorf("Could not initialize the storage: %v", err)
```

Comment:
```suggestion
			log.Errorf("Could not initialize in-memory storage: %v", err)
```

Reply:
Maybe another error message if no persistence storage is explicitly set?

Reply:
I copy/pasted this snippet from the other services just to be consistent. I would suggest not to change that in this PR.